Vic2ToHoI4/Source/HOI4World/HoI4Country.cpp

```diff
@@ -874,4 +874,71 @@ std::optional<HoI4FocusTree> HoI4::Country::getNationalFocus() const
 	{
 		return nullopt;
 	}
+}
+
+void HoI4::Country::addSphereling(std::string Sphereling)
+{
+	double defaultAutonomy = 0.5;
+	spherelings.insert(make_pair(Sphereling, defaultAutonomy));
+}
+
+double HoI4::Country::getSpherelingAutonomy(std::string spherelingTag) const
+{
+	auto sphereling = spherelings.find(spherelingTag);
+	if (sphereling == spherelings.end())
+	{
+		LOG(LogLevel::Debug) << "Could not find " << sphereling->first << " in spherelings";
+		return 0.5;
+	}
+	else
+	{
+		return sphereling->second;
+	}
+}
+
+void HoI4::Country::setSpherelingAutonomy(std::string spherelingTag, double autonomy)
+{
+	auto sphereling = spherelings.find(spherelingTag);
+	if (sphereling == spherelings.end())
+	{
+		LOG(LogLevel::Debug) << "Could not find " << sphereling->first << " in spherelings";
+		sphereling->second = 0.5;
+	}
+	else
+	{
+		sphereling->second = autonomy;
+	}
+}
+
+double HoI4::Country::calculateInfluenceFactor()
+{
+	auto GPInfluences = getGPInfluences();
+	if (sphereLeader.empty())
+	{
+		LOG(LogLevel::Debug) << tag << " has no sphere leader set";
+		return 55.5;
+	}
+	else
+	{
+		double influenceFactor = 0;
+		for (auto& influenceItr: GPInfluences)
+		{
+			if (influenceItr.first != sphereLeader)
```

Comment:
Wait, so influence from non-sphere leaders causes an increase, and influence from sphere leaders a decrease? This seems backwards.

Reply:
I never really elaborated the conversion process.
Sphere leader uses influence to reduce other GPs' influence (ban embassy, discredit, decrease relations), to prevent them from removing the sphereling from leader's sphere (or from ever getting to Friendly attitude). So I sum up all the influence from GPs already at Friendly attitude (ones that can actually remove the sphereling from sphere) and subtract 1.5 of Leader's influence.

This is the formula I use:
spherelingAutonomy = 3.6 * (Σ non-Leader Influence - 1.5 * Leader Influence) / 400

The idea is if all other GPs have 100+ differential the autonomy level is set to 90%, meaning it shouldn't be long before the sphereling breaks away.

Reply:
Gotcha!

A better name for the function might make this clear. Though I'm struggling to come up with one right now.

At the very least, you should put a comment up top summarizing this. Mention that this is outside influence that increases the sphereling's autonomy.

Reply:
> //Calculates Influence Factor = Σ Outside Influence - 1.5 * Leader Influence